packages/apm/src/integrations/tracing.ts

```diff
@@ -336,11 +343,171 @@ export class Tracing implements Integration {
+            addRequest(transactionSpan, entry);
+            break;
+          case 'mark':
+          case 'paint':
+          case 'measure':
+            const mark = transactionSpan.child({
+              description: `${entry.entryType} ${entry.name}`,
+              op: 'mark',
+            });
+            mark.startTimestamp = transactionSpan.startTimestamp + startTime - navigationOffset;
+            mark.timestamp = mark.startTimestamp + duration;
+            if (tracingInitMarkStartTime === undefined && entry.name === 'sentry-tracing-init') {
+              tracingInitMarkStartTime = mark.startTimestamp;
+            }
+            addSpan(mark);
+            break;
+          case 'resource':
+            const resourceName = entry.name.replace(window.location.origin, '');
+            if (entry.initiatorType === 'xmlhttprequest' || entry.initiatorType === 'fetch') {
+              // We need to update existing spans with new timing info
+              // tslint:disable-next-line: no-non-null-assertion
+              transactionSpan.spanRecorder!.finishedSpans.map((finishedSpan: SpanClass) => {
+                if (finishedSpan.description && finishedSpan.description.indexOf(resourceName) !== -1) {
+                  finishedSpan.startTimestamp = transactionSpan.startTimestamp + startTime - navigationOffset;
+                  finishedSpan.timestamp = finishedSpan.startTimestamp + duration;
+                }
+              });
+            } else {
+              const resource = transactionSpan.child({
+                description: `${entry.initiatorType} ${resourceName}`,
+                op: `resource`,
+              });
+              resource.startTimestamp = transactionSpan.startTimestamp + startTime - navigationOffset;
+              resource.timestamp = resource.startTimestamp + duration;
+              // We remeber the entry script end time to calculate the difference to the first init mark
+              if (
+                entryScriptStartEndTime === undefined &&
+                entryScriptSrc &&
+                entryScriptSrc.indexOf(resourceName) > -1
+              ) {
+                entryScriptStartEndTime = resource.timestamp;
+              }
+              addSpan(resource);
+            }
+            break;
+          default:
+          // Yo
+        }
+      });
+
+    if (entryScriptStartEndTime !== undefined && tracingInitMarkStartTime !== undefined) {
+      const evaluation = transactionSpan.child({
+        description: 'evaluation',
+        op: `script`,
+      });
+      evaluation.startTimestamp = entryScriptStartEndTime;
+      evaluation.timestamp = tracingInitMarkStartTime;
+      addSpan(evaluation);
+    }
+    if (global.performance) {
+      performance.clearMarks();
+      performance.clearMeasures();
+      performance.clearResourceTimings();
+      Tracing._performanceCursor = Math.max(performance.getEntries().length - 1, 0);
```

Comment:
@rhcarvalho From what I understand, it's to skip over certain entries that were created after clearing the performance entries.

Reply:
Correct, since not all of the performance entries can be cleared with `clear*()` we need to remember which entries we already considered.  And we call clear in the first place since browsers have an upper limit so this can overflow. TBH I am not sure if clearing this is a good practice but we'll see.